### PR TITLE
Bugfix/flow 10689 fast channel switch

### PR DESCRIPTION
**Pull request by lmancel**
ffmpeg pipelne rework for more stability

### CODE REVIEW

File: src/frontend/src/players/pipelines/ffmpegPlayingPipeline.ts [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+const ERROR = new PipelineState(PipelineStateName.ERROR)
+
+const PlayingPipelineStates = { STARTING, RETRIEVING_METADATA, RETRIEVED_METADATA, PLAYING, KILLED, ERROR }
+
+const transitions = {
+  [PipelineStateName.STARTING]: [RETRIEVING_METADATA, ERROR],
+  [PipelineStateName.RETRIEVING_METADATA]: [RETRIEVED_METADATA, KILLED, ERROR],
+  [PipelineStateName.RETRIEVED_METADATA]: [PLAYING, ERROR],
+  [PipelineStateName.PLAYING]: [KILLED, ERROR],
+  [PipelineStateName.KILLED]: [] as [],
```
Comment (Alexandre1624) [inline src/frontend/src/players/pipelines/ffmpegPlayingPipeline.ts]: [] as [], this is normal ?

File: src/frontend/src/players/player.ts [hunk 2]
```diff
@@ -9,9 +9,14 @@ import { IPipelineTail } from '../interfaces/playerPipeline'
 import { AbstractPlayer, PlayProps, SubtitlesProps } from './abstractPlayer'
 import { Store } from '../store'
 
-export class Player extends AbstractPlayer {
-  private replayOnErrorTimeout: number | null = null
+type Yo = {
```
Comment (Azevedo-Dani) [inline src/frontend/src/players/player.ts]: Yo ?

Comment (lmancel) [reply src/frontend/src/players/player.ts]: damn :p

File: src/frontend/src/players/pipelines/ffmpegPlayingPipeline.ts (hunk 1 above)
Comment (lmancel) [reply src/frontend/src/players/pipelines/ffmpegPlayingPipeline.ts]: :'(

image

Comment (lmancel) [reply src/frontend/src/players/pipelines/ffmpegPlayingPipeline.ts]: fixed

File: src/frontend/src/players/player.ts (hunk 2 above)
Comment (Alexandre1624) [inline src/frontend/src/players/player.ts]: Yo ?

Comment (lmancel) [reply src/frontend/src/players/player.ts]: cf https://github.com/taktik/flowr-desktop/pull/50#discussion_r984286406

**Review by Alexandre1624 — APPROVED** [review]
all comment has been resolve before I put as "need to work" :')

**Review by Azevedo-Dani — APPROVED** [review]

**Pull request merged**